### PR TITLE
Dwedul/1699 exchange keeper tests

### PR DESCRIPTION
**Pull request by SpicyLemon**
## Description

closes: #1699 

This PR:
* Add the (empty) `saffron-rc2` upgrade.
* Adds unit tests on the exchange keeper methods.
* During `FillBids`, the seller settlement fee is now calculated on the total price instead of each order individually.
* In the `OrderFeeCalc` query, ensure the market exists.
* In several endpoints, prevent the addresses from being converted to/from string/`AccAddress` when one or the other isn't needed.
* Fixes `InitGenesis`, to ensure `LastOrderId` is at least the largest order id.
* Properly populate the permissions lists when reading access grants from state.
* Fixes the paginated order queries to properly read orders from state.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [x] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: x/exchange/keeper/grpc_query_test.go [hunk 1]
```diff
@@ -1,27 +1,4017 @@
...
+	for _, tc := range tests {
+		s.Run(tc.name, func() {
+			runQueryTestCase(s, testDef, tc)
+		})
+	}
+}
+
+func (s *TestSuite) TestQueryServer_GetAllOrders() {
+	testDef := queryTestDef[exchange.QueryGetAllOrdersRequest, exchange.QueryGetAllOrdersResponse]{
+		queryName: "GetAllOrders",
```
Comment (Taztingo) [inline x/exchange/keeper/grpc_query_test.go:2941]: How large can the order list get? Will this become unusable without pagination after a certain length?

File: x/exchange/keeper/grpc_query_test.go [hunk 2]
```diff
@@ -1,27 +1,4017 @@
...
+
+func (s *TestSuite) TestQueryServer_GetAllMarkets() {
+	briefIDStringer := func(brief *exchange.MarketBrief) string {
+		if brief == nil {
+			return "<nil>"
+		}
+		return fmt.Sprintf("%d", brief.MarketId)
+	}
+	testDef := queryTestDef[exchange.QueryGetAllMarketsRequest, exchange.QueryGetAllMarketsResponse]{
+		queryName: "GetAllMarkets",
```
Comment (Taztingo) [inline x/exchange/keeper/grpc_query_test.go:3242]: I doubt this has the same issue as orders, but will this always be usable without pagination?

File: x/exchange/keeper/grpc_query_test.go (hunk 1 above)
Comment (SpicyLemon) [reply x/exchange/keeper/grpc_query_test.go:2941]: It is paginated. And yes, it'd be unusable at some length. But that exact length would depend on the processing power available to the node fielding the query.

The default limit is 100, which is used if no pagination parameters are set in the query, but there's no hard-coded maximum on the `limit`. If it's too high, the node will time-out the query after a few seconds.

File: x/exchange/keeper/grpc_query_test.go (hunk 2 above)
Comment (SpicyLemon) [reply x/exchange/keeper/grpc_query_test.go:3242]: This one also has pagination on it already. It'll probably be a long time until there's enough markets for it to be unusable without pagination, but any time there's a variable number of results in a query, it's best to include pagination.

**Review by Taztingo — APPROVED** [review]

**Review by iramiller — APPROVED** [review]

**Pull request merged**